src/glados/engine/tui_integration.py

```
@@ -0,0 +1,193 @@
+"""TUI integration mixin for the GLaDOS engine.
+
+This module provides the TUIIntegrationMixin class that extends the GLaDOS
+engine with capabilities to communicate with the TUI interface, sending
+real-time updates and notifications.
+"""
+
+import time
+from typing import Callable, Dict, Any
```

Ruff (0.11.9)

Import from `collections.abc` instead: `Callable`

Import from `collections.abc`

(UP035)